Makefile

```diff
@@ -124,6 +128,26 @@ e2e-helm-deploy:
 		./.staging/helm/linux-amd64/helm install manifest_staging/charts/gatekeeper --name-template=gatekeeper --set image.repository=${HELM_REPO} --set image.release=${HELM_RELEASE} --set emitAdmissionEvents=true --set emitAuditEvents=true;\
 	fi;
 
+e2e-helm-upgrade-init: e2e-helm-install
+	@if [ $$(echo ${HELM_VERSION} | head -c 1) = "2" ]; then\
+		kubectl create clusterrolebinding tiller-admin --clusterrole=cluster-admin --serviceaccount=kube-system:default;\
+		./.staging/helm/linux-amd64/helm init --wait --history-max=5;\
+		kubectl -n kube-system wait --for=condition=Ready pod -l name=tiller --timeout=300s;\
+		./.staging/helm/linux-amd64/helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts;\
+		./.staging/helm/linux-amd64/helm install gatekeeper/gatekeeper \
+			--version ${BASE_RELEASE} --name gatekeeper \
+			--set emitAdmissionEvents=true --set emitAuditEvents=true --wait;\
+	else\
+		./.staging/helm/linux-amd64/helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts;\
+		./.staging/helm/linux-amd64/helm install gatekeeper/gatekeeper \
+			--version ${BASE_RELEASE} --name-template gatekeeper \
+			--set emitAdmissionEvents=true --set emitAuditEvents=true --wait;\
+	fi;
+
+e2e-helm-upgrade:
```

Comment:
Is there anything specific we want to verify after the upgrade is done? e.g. image tag is greater than `BASE_RELEASE`

Since this is in-place upgrade, do we want to test how existing GK resources function in addition to new constraint templates and constraints? ie maybe we do not clean up all the test resources?

Reply:
right now, existing resources are cleaned up and e2e is run again from scratch. do we prefer keeping the same resources and just verifying them again? this means we won't be able to clean up after initial e2e and new bats test to verify existing resources with no new deployments.

Reply:
I’m ok with merging this PR as is, but we should create a followup PR that specifically tests how GK functions after the upgrade with exisitng gk resources in the cluster. So that could mean a new bats test file. wdyt?

Reply:
sounds good, we can discuss requirements for that as a follow-up

Reply:
Should theoretically have the same behavior... there's not much reason that adding a resource after a pod starts is different than adding a resource before the pod starts other than the bootstrapping logic.

There is the possibility of unanticipated behavior and maybe unanticipated enforcement of constraints against newly-introduced resources, however.